chainer/links/connection/multihead_attention.py

```diff
@@ -0,0 +1,198 @@
+            if not nobias:
+                self.proj_in_b = variable.Parameter(
+                    _initial_bias, (3 * self.embedding_size,))  # type: variable.Variable  # NOQA
+            else:
+                self.proj_in_b = None
+
+            self.out_proj = links.Linear(
+                self.embedding_size, self.embedding_size,
+                initialW=_initialW, initial_bias=_initial_bias, nobias=nobias)
+            self.proj_out_W = self.out_proj.W
+            self.proj_out_b = self.out_proj.b
+
+            if not nobias_kv:
+                self.bias_k = variable.Parameter(
+                    _initial_bias, (self.embedding_size,))  # type: variable.Variable  # NOQA
+                self.bias_v = variable.Parameter(
+                    _initial_bias, (self.embedding_size,))  # type: variable.Variable  # NOQA
+            else:
+                self.bias_k, self.bias_v = None, None
+
+    def forward(
+        self,
+        query,                     # type: tp.Optional[InputType]
+        key=None,                  # type: tp.Optional[InputType]
+        value=None,                # type: tp.Optional[InputType]
+        key_padding_mask=None,     # type: tp.Optional[InputType]
+        attention_mask=None,       # type: tp.Optional[InputType]
+        add_zero_attention=False,  # type: tp.Optional[bool]
+        return_weights=False       # type: tp.Optional[bool]
+    ):
+        # type: (...) -> tp.Union[tp.Tuple[variable.Variable, variable.Variable], variable.Variable]  # NOQA
+        """Compute attention weight and context vector.
+
+        Self-attention can be implemented by passing the same arguments for
+        query, key, and value. Timesteps can be masked by
+        giving a time x time mask in the `attention_mask`. Padding elements
+        can be excluded from the key by passing a batch_size x source_length
+        mask where padding elements are indicated by 1.
+
+        Args:
+            query (:class:`~chainer.Variable` or :ref:`ndarray`):
+                The query vectors with the shape of
+                (time, batch_size, query_in_size).
+            key (:class:`~chainer.Variable` or :ref:`ndarray`):
+                The key of the memory vectors with the shape of
+                (time, batch_size, key_in_size).
+            value (:class:`~chainer.Variable` or :ref:`ndarray`):
+                The value of the memory with the shape of
+                (time, batch_size, value_in_size).
+            key_padding_mask (:class:`~chainer.Variable` or :ref:`ndarray`):
+                If not ``None``, mask the memory slots.
+                The shape is (batch_size, source_length).
+                Each value is 0 or 1 where 1 means that
+                the memory slot will not be used.
+            attention_mask (:class:`~chainer.Variable` or :ref:`ndarray`)
+            return_weights (bool):
+                If ``True``, return both attention and attention weights.
+        Returns:
```

Comment:
Fix this by mimicking https://github.com/chainer/chainer/blob/master/chainer/functions/connection/n_step_rnn.py#L443

> 
    Returns:
        tuple: This function returns a tuple containing three elements,
        ``hy`` and ``ys``.
        - ``hy`` is an updated hidden states whose shape is same as ``hx``.
        - ``ys`` is a list of :class:`~chainer.Variable` . Each element
          ``ys[t]`` holds hidden states of the last layer corresponding
          to an input ``xs[t]``. Its shape is ``(B_t, N)`` where ``B_t`` is
          mini-batch size for time ``t``, and ``N`` is size of hidden
          units. Note that ``B_t`` is the same value as ``xs[t]``.